File: 프로그래머스/2/42587. 프로세스/프로세스.js
function solution(priorities, location) {
    var answer = 0;
    
    let check = priorities.map((a, idx) => {
        if(idx === location)
            return true;
        return false;
    });
    
    let max = Math.max.apply(null, priorities);
    while(1){
        if(priorities[0] < max){
            priorities.push(priorities[0]);
            priorities.shift();
            check.push(check[0]);
            check.shift();
        }
        else{
            if(check[0] === true){
                answer++;
                break;
            }  
            
            priorities.shift();
            check.shift();
            answer++;
            
            max = Math.max.apply(null, priorities);
        }
            
    }
    
    return answer;
}

File: 프로그래머스/2/42587. 프로세스/프로세스.cpp
#include <string>
#include <vector>
#include <queue>

using namespace std;

int solution(vector<int> priorities, int location) {
    int answer = 0;
    
    queue<pair<int, bool>> q;
    priority_queue<int> pq;

    for (int i = 0; i < priorities.size(); i++) {
        if (i == location) {
            q.push({ priorities[i], true });
            pq.push(priorities[i]);
            continue;
        }

        q.push({ priorities[i], false });
        pq.push(priorities[i]);
    }

    int max = pq.top();
    while (1) {
        if (q.front().first < max) {
            q.push(q.front());
            q.pop();
        }
        else {
            if (q.front().second) {
                answer++;
                break;
            }

            q.pop();
            pq.pop();
            answer++;

            max = pq.top();
        }
    }
    
    return answer;
}